private/pkg/git/repository.go

```diff
@@ -125,17 +133,20 @@ func (r *repository) ForEachBranch(f func(string, Hash) error) error {
 		}
 		unpackedBranches[branchName] = struct{}{}
 		return f(branchName, hash)
-	}); err != nil {
+	}); err != nil && !errors.Is(err, fs.ErrNotExist) {
```

Comment:
If the `branchesDir` does not exist (like someone asking for branches in a remote name that does not exist), instead of failing, we just don't loop over any branch.

Also, it might be possible that we do have some packed branches for that repo.